source/fem-dem/vans_assemblers.cc

```diff
@@ -1260,40 +1257,35 @@ GLSVansAssemblerGidaspow<dim>::calculate_particle_fluid_interactions(
 
       cell_void_fraction = scratch_data.cell_void_fraction[particle_number];
 
-      // Gidaspow Drag Model CD Calculation
-      if (Re_p[particle_number] < 1000)
-        {
-          C_d = 24 / Re_p[particle_number] *
-                (1 + 0.15 * pow(Re_p[particle_number], 0.687));
-        }
-      else
-        {
-          C_d = 0.44;
-        }
+      // Gidaspow Drag Model
+      double particle_density =
+        particle_properties[DEM::PropertiesIndex::mass] /
+        (M_PI * pow(particle_properties[DEM::PropertiesIndex::dp], dim) /
+         (2.0 * dim));
 
-      if (cell_void_fraction >= 0.8)
+      if (cell_void_fraction > 0.8)
         {
           momentum_transfer_coefficient =
-            0.75 * C_d * cell_void_fraction * average_relative_velocity.norm() *
-            (1 - cell_void_fraction) * pow(cell_void_fraction, -2.65) /
-            particle_properties[DEM::PropertiesIndex::dp];
+            (18 * pow(cell_void_fraction, -3.65) *
+             (1 + 0.15 * pow(Re_p[particle_number], 0.687))) *
+            (particle_properties[DEM::PropertiesIndex::mass] * viscosity /
+             (pow(particle_properties[DEM::PropertiesIndex::dp], 2) *
+              particle_density));
         }
       else
         {
           // Assuming the sphericity of particles = 1
           momentum_transfer_coefficient =
-            150 * pow((1 - cell_void_fraction), 2) * viscosity /
-              (cell_void_fraction *
-               pow(particle_properties[DEM::PropertiesIndex::dp], 2)) +
-            1.75 * (1 - cell_void_fraction) * average_relative_velocity.norm() /
-              particle_properties[DEM::PropertiesIndex::dp];
+            (150 * (1 - cell_void_fraction) / pow(cell_void_fraction, 2) +
```

Comment:
might be a good idea to use instead https://www.dealii.org/current/doxygen/deal.II/namespaceUtilities.html#a834c6b595ada6f8e73c78d9bbd8ec551
when you have fixed powers